File: src/main/java/cz/kamenitxan/jakon/utils/validation/validators/NotEmptyValidator.java
package cz.kamenitxan.jakon.utils.validation.validators;

import cz.kamenitxan.jakon.utils.validation.Validator;

import java.lang.reflect.Array;
import java.util.Collection;
import java.util.Map;

/**
 * Created by TPa on 11.08.16.
 */
public class NotEmptyValidator implements Validator {
	@Override
	public boolean isValid(Object... params) {
		if (params == null ) {
			return false;
		}

		final Object value = params[0];
		if (value.getClass().isArray()) {
			return Array.getLength(value) > 0;
		}
		if (value instanceof Collection) {
			return !((Collection<?>) value).isEmpty();
		}
		if (value instanceof Map) {
			return !((Map<?, ?>) value).isEmpty();
		}
		if (value instanceof String) {
			return !((String) value).trim().isEmpty();
		}
		return true;
	}
}

File: modules/core/src/main/scala/cz/kamenitxan/jakon/validation/validators/NotEmptyValidator.scala
package cz.kamenitxan.jakon.validation.validators


import cz.kamenitxan.jakon.validation.{ValidationResult, Validator}

import java.lang.annotation.Annotation
import java.lang.reflect.Field

class NotEmptyValidator extends Validator {
	private val error = "EMPTY"

	override def isValid(value: String, a: Annotation, field: Field, data: Map[Field, String]): Option[ValidationResult] = {
		if (value == null) {
			return ValidationResult(error)
		}
		if (value.nonEmpty) {
			Option.empty
		} else {
			ValidationResult(error)
		}
	}
}
